### ISSUE TITLE
Commit disappeared from the master branch

### ISSUE THREAD

**Issue by matthias-mayr** [luinardi/hypermapper#74]
I have a research project that uses `hypermapper` as a Git submodule. Git submodules always point to a specific commit. In my case the current pointer was to the commit `076067c2723e5bf33bcf7bcb0b02deab7415c691`:
```
commit 076067c2723e5bf33bcf7bcb0b02deab7415c691 (HEAD -> master)
Author: arturluis <arturluis94@gmail.com>
Date:   Wed Sep 15 18:22:14 2021 -0300

    Updated gaussian prior interface and added cma-es.
```
Especially in research projects in can be very helpful to keep dependencies at a specific version. One of the main reasons is the reproducibility of research results.

Cloning this project today it looks like this commit has disappeared from the master branch. Could that have happened as part of a rebase or forced push?

**Comment by arturluis** [comment]
Hi @matthias-mayr ,

As far as I know, there was no rebase or force push, we indeed avoid doing those. I was checking the commit log here and I still see that commit, but under a different hash:
![image](https://user-images.githubusercontent.com/5957286/167208200-f94d675a-46c0-4e10-b4a0-7bb99cab1bd5.png)

which matches the hash I have in the repo copies I have in my computer:

![image](https://user-images.githubusercontent.com/5957286/167208234-e6a1be7e-e936-42f4-81ac-e5680124ccc7.png)

I am not sure what could have caused this mismatch. Do you remember the date you created the submodule in your project? That might help find out how or why the hash changed.

**Comment by matthias-mayr** [comment]
Thanks for the quick reply.

**tl;dr:** totally my fault. I cherry-picked that commit.

**Longer story:** I had issues with the `custom_gaussians` in the json file. It did not pass the schema verification. Cherry picking that commit from the future on top of v2.2.4 resolved it as a quick fix before a deadline.  
Not sure how this cherry-picked commit ID appeared in this other branch. That should not have happened.

In any case: thanks for the quick clarification.
